designer/client/conditions/SelectConditions.tsx

```diff
@@ -75,141 +71,113 @@ class SelectConditions extends React.Component<Props, State> {
     const inputs = path
       ? inputsAccessibleAt(data, path)
       : allInputs(data) ?? [];
-    return inputs
-      .map((input) => ({
-        label: input.title,
-        name: this.trimSectionName(input.propertyPath),
-        type: input.type,
-      }))
-      .reduce((obj, item) => {
-        obj[item.name] = item;
-        return obj;
-      }, {});
+    return inputs.map((input) => input.propertyPath);
   }
 
   conditionsForPath(path: string) {
     const { data } = this.context;
     const fields: any = Object.values(this.fieldsForPath(path));
     const { conditions = [] } = data;
-    let conditionsForPath: any[] = [];
-    const conditionsByTypeMap = conditionsByType(conditions);
-
-    fields.forEach((field) => {
-      this.handleStringConditions(
-        conditionsByTypeMap.string,
-        field.name,
-        conditionsForPath
-      );
-      this.handleConditions(
-        conditionsByTypeMap.object,
-        field.name,
-        conditionsForPath
-      );
-      this.handleNestedConditions(
-        conditionsByTypeMap.nested,
-        field.name,
-        conditionsForPath
-      );
+    let sortedConditions = this.sortConditions(conditions);
+    let validConditionNames = [];
+    return sortedConditions.filter((condition) => {
+      if (getConditionType(condition) === "string") {
+        return this.handleStringCondition(
+          condition,
+          fields,
+          validConditionNames
+        );
+      }
+      if (getConditionType(condition) === "nested") {
+        return this.handleNestedCondition(
+          condition,
+          fields,
+          validConditionNames
+        );
+      }
+      if (getConditionType(condition) === "object") {
+        return this.handleCondition(condition, fields, validConditionNames);
+      }
+      return false;
     });
+  }
 
-    return conditionsForPath;
+  sortConditions(conditions: ConditionData[]) {
+    return conditions
+      .reduce<ConditionData[][]>(
+        (acc, curr) => {
+          acc[getConditionType(curr) === "nested" ? 1 : 0].push(curr);
+          return acc;
+        },
+        [[], []]
+      )
+      .flat();
   }
 
-  handleConditions(
-    objectConditions: ConditionData[],
-    fieldName: string,
-    conditionsForPath: any[]
+  handleCondition(
+    condition: ConditionData,
+    fields: string[],
+    validConditions: string[]
   ) {
-    objectConditions.forEach((condition) => {
-      condition.value.conditions?.forEach((innerCondition) => {
-        this.checkAndAddCondition(
-          condition,
-          fieldName,
-          getFieldNameSubstring(innerCondition.field.name),
-          conditionsForPath
-        );
-      });
-    });
+    let conditionValue = condition.value as ConditionObject;
+    for (let i = 0; i < conditionValue.conditions.length; i++) {
+      if (!fields.includes(conditionValue.conditions[i].field.name)) {
+        return false;
+      }
+    }
+    validConditions.push(condition.name);
```

Comment:
There are a lot of side effects going on here that "work" but are unintended or difficult to follow. Anyone not familiar with javascript won't know what this is actually doing (validConditions being a reference - and updating something outside of scope).

For JS - especially JS frontend frameworks, ideally write fp style. It's not exactly relevant here but JS frameworks are usually checking if a reference has changed - and is able to re-render the page. If references don't change, e.g. you use arr.push() - it might not detect something has changed and therefore won't rerender.